debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalSnapshotChangeEventSource.java

```diff
@@ -216,6 +220,26 @@ private void delaySnapshotIfNeeded(ChangeEventSourceContext context) throws Inte
      */
     protected void connectionCreated(SnapshotContext snapshotContext) throws Exception {
     }
+    
+    private Optional<TableId> toTableId(Set<TableId> tableIds, Pattern id) {
```

Comment:
What if more than one table matches? IMHO you yhoul return stream of TableIds and then `flatMap` it